Python/import.c

```diff
@@ -249,16 +249,7 @@ import_get_module(PyThreadState *tstate, PyObject *name)
 
     PyObject *m;
     Py_INCREF(modules);
-    if (PyDict_CheckExact(modules)) {
-        m = PyDict_GetItemWithError(modules, name);  /* borrowed */
-        Py_XINCREF(m);
-    }
-    else {
-        m = PyObject_GetItem(modules, name);
-        if (m == NULL && _PyErr_ExceptionMatches(tstate, PyExc_KeyError)) {
-            _PyErr_Clear(tstate);
-        }
-    }
+    (void)_PyMapping_LookupItem(modules, name, &m);
```

Comment:
Out of interest, why not explicitly `_PyErr_Clear(tstate)` if `_PyMapping_LookupItem` returns `-1`?

Reply:
Because the current code does not clear arbitrary errors (and doing this would be wrong). It only clears `KeyError`, what is done internally by `_PyMapping_LookupItem`.